controllers/discovery/searchTopic.js

```diff
@@ -10,7 +10,10 @@ const {sequelize} = require('../../databases/models');
  */
 const Search = async (req, res) => {
   const {q} = req.query;
+
   const {userId} = req;
+  const anonymousUser = await UsersFunction.findAnonymousUserId(User, userId, {raw: true});
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The anonymous user ID is fetched using the `findAnonymousUserId` function from `UsersFunction`. This is a good approach, but there should be error handling in case the function fails or returns null.
```diff
+   const anonymousUser = await UsersFunction.findAnonymousUserId(User, userId, {raw: true});
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->